addons/website/static/src/snippets/s_website_form/000.js

```diff
@@ -141,10 +141,10 @@ odoo.define('website.s_website_form', function (require) {
                 // customizations that use the current behavior as a feature.
                 for (const name of fieldNames) {
                     const fieldEl = this.$target[0].querySelector(`[name="${name}"]`);
-                    if (!fieldEl.value && dataForValues && dataForValues[name]) {
+                    if (dataForValues && dataForValues[name]) {
                         fieldEl.value = dataForValues[name];
                         this.filledFields.push(fieldEl);
-                    } else if (!fieldEl.value && this.preFillValues[fieldEl.dataset.fillWith]) {
+                    } else if (this.preFillValues[fieldEl.dataset.fillWith]) {
```

Comment:
@Guillaume-gdi In the first commit, you add in `destroy` a reset of the value for the `filledFields`. You reset them via `.value = ''`. In this third commit, you allow prefill values to replace existing default values (not empty ones)... but on destroy you still do `.value = ''`. It works most of the time as `.value = ''` only writes on the property... but in the case of an hidden input with a configured default value, you probably still have the bug of losing the configured default value if it was prefilled first, is not it? :grimacing: 

Probably worth an extra check in the website_hr_recruitment test in this commit?